### DISCUSSION TITLE
where are the grammar files `fetch`ed and `buil[t]`?

### DISCUSSION THREAD

**Discussion by amtoine** [helix-editor/helix#4137] [Q&A] [answered]
hello there :wave: :yum: 

i could not find a discussion on that very topic, hope i'm not creating a duplicate here :relieved: 

## my current setup
i have switched from the `community` package on ArchLinux to the "build from source" method, mainly to avoid messing around with either `/var/lib/helix/` or `/usr/lib/helix/` which both require `sudo` privileges...
i've ran the following, according to the [documentation](https://docs.helix-editor.com/install.html#build-from-source) :ok_hand: 
```bash
git clone https://github.com/helix-editor/helix
cd helix
cargo install --path helix-term
```
now i have `hx` available in `$CARGO_HOME/bin/` :ok_hand: 

i've even been able to `cp` all the `runtime` to `~/.config/helix/runtime/` and (**1**) get back my `dracula` theme and (**2**) add `pascal`'s syntax highlighting for the `oberon` language as introduced in #4128 :star_struck: 
just had to run
```bash
cp -r ~/.config/helix/runtime/queries/pascal/ ~/.config/helix/runtime/queries/oberon/
```

## my actual question
everything seems to work fine for now but i do not understand the following
```bash
> hx --grammar fetch; hx --grammar build
Fetching 114 grammars
114 up to date git grammars
Building 114 grammars
114 grammars already built
```

**where are these grammars files fetched and built to?** :thinking: 

i could not find any "helix" directory in `/usr/share/`, `/usr/lib/` nor `/var/lib/`
nothing seems to have changed in my `~/.config/helix/runtime/`
it gives me the same output even when removing the source repository :confused:

**Comment by Ordoviz** [comment, marked as answer]
You can find  the compiled tree-sitter grammars in `~/.config/helix/runtime/grammars`. Are you asking out of curiosity or do you want to solve a specific problem?

**Reply by amtoine** [reply]
mm actually, it's both :yum: 

when building from source, we always need to `cp runtime ~/.config/helix/runtime -r`?
then `hx --grammar fetch; hx --grammar build`
a little `hx --health` to check everything is right

and everything is done in `~/.config/helix/runtime/`?

**Reply by amtoine** [reply]
is it possible to see the paths `helix` is using when invoking `hx --grammar` for instance? :open_mouth: 
i think that would be really handy to debug what is happening :yum:

**Reply by Ordoviz** [reply]
>is it possible to see the paths helix is using when invoking `hx --grammar` for instance?

You can prepend `strace` to any command to see which syscalls it is making (interacting with the file system requires syscalls). You might have to `pacman -S strace` first.

**Reply by amtoine** [reply]
oooooh that is nice to know :open_mouth: 
thanks :yum:

**Reply by amtoine** [reply]
i was more thinking about some output in `stdout` :thinking: 

for instance, print `xxx / xxx: fetching grammar for xxx` for each language and the same with building
and maybe giving the path used direcly in the output of both `grammar fetch` and `grammar build` :confused: 
'cause `strace` is full of information, but honestly quite hard to read for beginners and on a daily basis when trying to fetch and build some grammars :open_mouth: 

**would the `helix` team be open for such a feature?** :yum: 
i can open an issue and even try to dig down the source base to see if i can help :relieved:

**Reply by the-mikedavis** [reply]
I don't think it's necessary, `hx --health` shows the runtime directory

**Reply by amtoine** [reply]
> I don't think it's necessary, `hx --health` shows the runtime directory
>
oooooh you are 100% right :star_struck: 

i did not see that line :eyes: 
then it is perfect for me :relieved: 

thanks for the information :wink:
